client/components/review.js

```diff
@@ -149,6 +174,6 @@ class Review extends React.Component {
   }
 }
 
-const mapStateToProps = ({ project, settings, application: { readonly } }) => ({ project, settings, readonly });
+const mapStateToProps = ({ project, settings, application: { readonly }, changes : {latest, granted} }) => ({ project, settings, readonly, latest, granted });
```

Comment:
It might be better to traverse the changes in this function using the props provided and _only_ pass the changed state for this one question rather than the entire list.

The reasoning being that if the entire list is passed then this field will re-render whenever any change is made, not just the one field that we care about.

Having said that, I don't think it matters _too much_ here because it's static data, but worth considering for future components.

Reply:
```js
const mapStateToProps = ({ project, settings, application: { readonly }, changes : {latest, granted} }, { name, prefix }) => {
  const key = `${this.props.prefix || ''}${this.props.name}`;
  const changedFromGranted = granted.includes();
  const changedFromLatest = latest.includes();
  return { project, settings, readonly, changedFromLatest, changedFromGranted };
}
```

Reply:
thank you